src/providers/ldap/ldap_id_subid.c

```diff
@@ -141,6 +150,102 @@ static void subid_ranges_get_connect_done(struct tevent_req *subreq)
         return;
     }
 
+    subid_ranges_resolve_owner(req);
+}
+
+static char *get_user_dn(TALLOC_CTX *mem_ctx,
+                         struct sss_domain_info *domain,
+                         const char* name)
+{
+    const char *attrs[] = {SYSDB_ORIG_DN, SYSDB_CACHE_EXPIRE, NULL};
+    struct ldb_result *res = NULL;
+    time_t expire;
+    char *result = NULL;
+    const char *dn;
+    int ret;
+    char *fqdn = sss_create_internal_fqname(mem_ctx, name, domain->name);
+    if (fqdn == NULL) {
+        return NULL;
+    }
+
+    ret = sysdb_get_user_attr(fqdn, domain, fqdn, attrs, &res);
+    if ((ret == EOK) && (res != NULL) && (res->count == 1)) {
+        expire = ldb_msg_find_attr_as_uint64(res->msgs[0], SYSDB_CACHE_EXPIRE, 0);
+        if (expire > time(NULL)) {
+            dn = ldb_msg_find_attr_as_string(res->msgs[0], SYSDB_ORIG_DN, NULL);
+            if (dn != NULL) {
+                sss_filter_sanitize(mem_ctx, dn, &result);
```

Comment:
Covscan complains here:
```
Calling sss_filter_sanitize without checking return value (as is done elsewhere 66 out of 67 times).
```
But when functions fails its output arg is untouched, and thus NULL. Not much to do on top of this.